testnet/stacks-node/src/tests/epoch_22.rs

```diff
@@ -0,0 +1,1192 @@
+        Some(AddressHashMode::SerializeP2PKH),
+    );
+    let pox_addr_3 = PoxAddress::Standard(
+        StacksAddress::new(26, Hash160::from_hex(&pox_pubkey_hash_3).unwrap()),
+        Some(AddressHashMode::SerializeP2PKH),
+    );
+    let burn_pox_addr = PoxAddress::Standard(
+        StacksAddress::new(
+            26,
+            Hash160::from_hex("0000000000000000000000000000000000000000").unwrap(),
+        ),
+        Some(AddressHashMode::SerializeP2PKH),
+    );
+
+    let expected_slots = HashMap::from([
+        (42u64, HashMap::from([(pox_addr_1.clone(), 4u64)])),
+        (43, HashMap::from([(pox_addr_1.clone(), 4)])),
+        (44, HashMap::from([(pox_addr_1.clone(), 4)])),
+        // cycle 45 is the first 2.1, and in the setup of this test, there's not
+        //  enough time for the stackers to begin in this cycle
+        (45, HashMap::from([(burn_pox_addr.clone(), 4)])),
+        (46, HashMap::from([(burn_pox_addr.clone(), 4)])),
+        (
+            47,
+            HashMap::from([(pox_addr_2.clone(), 2), (pox_addr_3.clone(), 2)]),
+        ),
+        // Now 2.2 is active, everything should be a burn.
+        (48, HashMap::from([(burn_pox_addr.clone(), 4)])),
+        (49, HashMap::from([(burn_pox_addr.clone(), 4)])),
+        (50, HashMap::from([(burn_pox_addr.clone(), 4)])),
+    ]);
+
+    for reward_cycle in reward_cycle_min..(reward_cycle_max + 1) {
+        let cycle_counts = match reward_cycle_pox_addrs.get(&reward_cycle) {
+            Some(x) => x,
+            None => {
+                info!("No reward cycle entry = {}", reward_cycle);
+                continue;
+            }
+        };
+        assert_eq!(cycle_counts.len(), expected_slots[&reward_cycle].len(), "The number of expected PoX addresses in reward cycle {} is mismatched with the actual count.", reward_cycle);
+        for (pox_addr, slots) in cycle_counts.iter() {
+            assert_eq!(
+                *slots,
+                expected_slots[&reward_cycle][&pox_addr],
+                "The number of expected slots for PoX address {} in reward cycle {} is mismatched with the actual count.",
+                &pox_addr,
+                reward_cycle,
+            );
+            info!("PoX payment received"; "cycle" => reward_cycle, "pox_addr" => %pox_addr, "slots" => slots);
+        }
+    }
+
+    let blocks = test_observer::get_blocks();
+    for block in blocks {
+        let transactions = block.get("transactions").unwrap().as_array().unwrap();
+        for tx in transactions {
+            let raw_tx = tx.get("raw_tx").unwrap().as_str().unwrap();
+            if raw_tx == "0x00" {
+                continue;
+            }
+            let tx_bytes = hex_bytes(&raw_tx[2..]).unwrap();
+            let parsed =
+                StacksTransaction::consensus_deserialize(&mut tx_bytes.as_slice()).unwrap();
+            let tx_sender = PrincipalData::from(parsed.auth.origin().address_testnet());
```

Comment:
Is there something you wanted to check here?

Reply:
Not specifically, I left the tx receipt parsing in the test in case I did want to check the transactions, but I check the transaction success by asserting the transfers altered the balances.